libs/sysplugins/smarty_resource_custom.php

```diff
@@ -88,6 +88,6 @@ public function getContent(Smarty_Template_Source $source)
      */
     public function getBasename(Smarty_Template_Source $source)
     {
-        return basename(substr(preg_replace('/[^A-Za-z0-9.]/', '', $source->name), 0, 25));
+        return basename(substr(preg_replace('/[^A-Za-z0-9._]/', '', $source->name), 0, 255));
```

Comment:
I'd wrap this transformation into a separate method.
You already using it in two places and the code like this is untestable.